From: Claudio Imbrenda <imbrenda@linux.vnet.ibm.com>
Subject: KVM: s390: Enable all facility bits that are known good for passthrough
Patch-mainline: v4.7-rc1
Git-commit: ed8dda0bf74b4fb8e73d8880c78effabd3285fd8
References: FATE#324071, LTC#158956, bsc#1073059

Summary:     kernel: Transparent Facility Bit Handling
Description: When a guest is initializing, KVM provides facility bits that can
             be successfully used by the guest, by applying a mask to the
             bitfield returned by the STFLE instruction, so that only the
             facility bits actually supported by KVM are passed to the guest.
             
             Some facilities need support in the hypervisor in order to work,
             while others are handled by the system and do not need explicit
             support from the hypervisor.
             
             Until now the mask to select the bits to pass to the guest was
             maintained manually; this patchset allows KVM to pass to the guest
             all the facility bits that do not require hypervisor support.
             This allows new hardware features that do not need hypervisor
             support to be available for guests without having to patch the
             hypervisor every time.

Upstream-Description:

             KVM: s390: Enable all facility bits that are known good for passthrough

             Some facility bits are in a range that is defined to be "ok for guests
             without any necessary hypervisor changes". Enable those bits.

             Signed-off-by: Alexander Yarygin <yarygin@linux.vnet.ibm.com>
             Reviewed-by: David Hildenbrand <dahi@linux.vnet.ibm.com>
             Signed-off-by: Christian Borntraeger <borntraeger@de.ibm.com>


Signed-off-by: Claudio Imbrenda <imbrenda@linux.vnet.ibm.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 arch/s390/kvm/kvm-s390.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/arch/s390/kvm/kvm-s390.c
+++ b/arch/s390/kvm/kvm-s390.c
@@ -120,8 +120,8 @@ struct kvm_stats_debugfs_item debugfs_en
 
 /* upper facilities limit for kvm */
 unsigned long kvm_s390_fac_list_mask[] = {
-	0xffe6fffbfcfdfc40UL,
-	0x005e800000000000UL,
+	0xffe6ffffffffffffUL,
+	0x005effffffffffffUL,
 	0x3000000000000000UL,
 };
 
